### PR TITLE
fix(web-components): adds short title functionality and prevents truncation

### PR DESCRIPTION
**Pull request by j212122**
## Summary of the changes
Adds functionality too allow long app title to be shortened on smaller viewports. Prevents truncation or ellipses of longer nav-items

## Related issue
#854 

## Checklist
- [x] I have added relevant unit and visual regression tests.

### CODE REVIEW

File: packages/web-components/src/components/ic-search-bar/readme.md [hunk 1]
```diff
@@ -68,13 +68,6 @@ Type: `Promise<void>`
 
 
 
-## CSS Custom Properties
-
-| Name                | Description                |
-| ------------------- | -------------------------- |
-| `--ic-z-index-menu` | z-index of search bar menu |
-
-
```
Comment (GCHQ-Developer-530) [inline packages/web-components/src/components/ic-search-bar/readme.md:77]: This needs to stay in, you can run npm run build and it should bring it back

**Review by GCHQ-Developer-530 — CHANGES_REQUESTED** [review]

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Review by GCHQ-Developer-465 — APPROVED** [review]

File: packages/web-components/.loki/reference/chrome_iphone7_Web_Components_Side_navigation_Short_title.png
Comment (jd239) [inline packages/web-components/.loki/reference/chrome_iphone7_Web_Components_Side_navigation_Short_title.png:1]: How comes the padding around the app name here is bigger than the long title?

File: packages/web-components/src/components/ic-side-navigation/test/basic/ic-side-navigation.e2e.ts [hunk 2]
```diff
@@ -459,6 +460,38 @@ describe("ic-side-navigation", () => {
       expect(applicationNameOpacity).toBe("1");
     });
 
+    it.only("should toggle to short title at small screen sizes", async () => {
```
Comment (jd239) [inline packages/web-components/src/components/ic-side-navigation/test/basic/ic-side-navigation.e2e.ts:463]:
```suggestion
    it("should toggle to short title at small screen sizes", async () => {
```

File: packages/web-components/src/components/ic-side-navigation/test/basic/ic-side-navigation.e2e.ts [hunk 3]
```diff
@@ -459,6 +460,38 @@ describe("ic-side-navigation", () => {
...
+
+      await page.waitForChanges();
+
+      appTitle = await page.find(
+        "ic-side-navigation >>> .title-link ic-typography"
+      );
+
+      expect(appTitle.textContent).toBe("ApplicationName");
+    });
+
```
Comment (jd239) [inline packages/web-components/src/components/ic-side-navigation/test/basic/ic-side-navigation.e2e.ts:494]: 👍

**Review by jd239 — CHANGES_REQUESTED** [review]
One minor change and a comment..

**Comment by jd239** [comment]
Noticed 2 issues:

1. Now that the navigation item label wraps, when opening an collapsed side nav (on md, lg and above), and then closing, because the width is reduced, the label squashes before it is fully faded out. An option is to force the width of the label to not collapse as the side nav collapses or, instead of a fade animation, it just disappears instantly

2. On collapsed icons with labels, when opening and closing the side nav, 'different navigation' gets cut off because the height and overflow is not aware of the wrapped label. 
Capture

**Review by jd239 — APPROVED** [review]

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Pull request merged**